lib/wigets/app_header.dart

```diff
@@ -0,0 +1,200 @@
+import 'dart:developer';
+
+import 'package:flutter/material.dart';
+import 'package:fokus/data/model/user/user_type.dart';
+import 'package:fokus/data/model/user/user.dart';
+import 'package:fokus/utils/app_locales.dart';
+import 'package:fokus/utils/theme_config.dart';
+
+enum AppHeaderType { greetings, normal }
+
+class HeaderActionButton {
+	final IconData icon;
+	final String text;
+	final Function action;
+	final Widget customContent;
+	final Color backgroundColor;
+
+	HeaderActionButton(this.icon, this.text, this.customContent, this.action, [this.backgroundColor]);
+	HeaderActionButton.normal(IconData icon, String text, Function action, [Color backgroundColor])
+		: this(icon, text, null, action, backgroundColor);
+	HeaderActionButton.custom(Widget customContent, Function action, [Color backgroundColor])
+		: this(null, null, customContent, action, backgroundColor);
+}
+
+class AppHeader extends StatelessWidget {
+	final User user;
+	final String title;
+	final String text;
+	final List<HeaderActionButton> headerActionButtons;
+	final AppHeaderType headerType;
+
+	AppHeader(this.user, this.title, this.text, this.headerActionButtons, this.headerType);
+	AppHeader.greetings(User user, String text, List<HeaderActionButton> headerActionButtons) 
+		: this(user, user.name, text, headerActionButtons, AppHeaderType.greetings);
+	AppHeader.normal(User user, String title, String text, List<HeaderActionButton> headerActionButtons) 
+		: this(user, title, text, headerActionButtons, AppHeaderType.normal);
```

Comment:
Not that it's a mistake but by convention widgets in Flutter (especially that big) use named parameters